apps/prairielearn/src/pages/studentAssessmentInstance/studentAssessmentInstance.js

```diff
@@ -9,14 +9,27 @@ const error = require('@prairielearn/error');
 const assessment = require('../../lib/assessment');
 const studentAssessmentInstance = require('../shared/studentAssessmentInstance');
 const sqldb = require('@prairielearn/postgres');
-var groupAssessmentHelper = require('../../lib/groups');
+const groupAssessmentHelper = require('../../lib/groups');
+const { AssessmentInstanceSchema } = require('../../lib/db-types');
 
 const sql = sqldb.loadSqlEquiv(__filename);
 
+async function ensureUpToDate(locals) {
+  const updated = await assessment.update(locals.assessment_instance.id, locals.authn_user.user_id);
+  if (!updated) return;
+
+  // we updated the assessment_instance, so reload it
+  const result = sqldb.queryRow(
+    sql.select_assessment_instance,
+    { assessment_instance_id: locals.assessment_instance.id },
+    AssessmentInstanceSchema,
+  );
+  locals.assessment_instance = result.rows[0];
+}
+
```

Comment:
Since this had to be added to exam I took the liberty of modernizing it.

Reply:
I had to revert the `queryRow` schema since `AssessmentInstanceSchema` is not declared yet. I want to minimize new features here, so I'm reverting to `queryAsync` for now.